ls1intum/Artemis
# `Programming exercises`: Show warning in build plan editor if build plan could not be found

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises
- [x] ~~I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~~ Not applicable, feature not available there.

### Motivation and Context
Closes #6784

### Description
Handles the error sent from the server and shows a warning to the user.

### Steps for Testing

Prerequisites:
- Gitlab/Jenkins setup (e.g. TS2)
- 1 Instructor
- 1 old Programming Exercise *created before 6.3.0*

1. Go the the programming exercise details page in the course management.
2. Button block at the top: → Edit build plan
3. A warning should be shown.
4. Create a new programming exercise.
5. Go to the Edit build plan again, this time no warning is shown and the editor is filled with content.


### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| `build-plan-editor.component.ts` | 98% | ✅ |


### Screenshots
n/a
